demos/burgers-isotropic.py

```diff
@@ -0,0 +1,274 @@
+# all timesteps.
+#
+# Note that when adapting the mesh, we need to be careful to check whether convergence
+# has already been reached on any of the subintervals. If so, the adaptation step is
+# skipped. ::
+
+
+def adaptor(mesh_seq, solutions, indicators):
+    metrics = []
+    complexities = []
+
+    # Ramp the target average metric complexity per timestep
+    base, target, iteration = 400, 1000, mesh_seq.fp_iteration
+    mp = {
+        "dm_plex_metric": {
+            "target_complexity": ramp_complexity(base, target, iteration),
+            "p": 1.0,
+            "h_min": 1.0e-04,
+            "h_max": 1.0,
+        }
+    }
+
+    # Construct the metric on each subinterval
+    for i, mesh in enumerate(mesh_seq):
+        sols = solutions["u"]["forward"][i]
+        dt = mesh_seq.time_partition.timesteps[i]
+
+        # Define the Riemannian metric
+        P1_ten = TensorFunctionSpace(mesh, "CG", 1)
+
+        # At each timestep, recover metric of the solution
+        # vector. Then time integrate over the contributions
+        _metrics = []
+
+        for j, sol in enumerate(sols):
+
+            # get local indicator
+            indi = indicators["u"][i][j]
+            # local instance of Riemanian metric
+            _metric = RiemannianMetric(P1_ten)
+
+            # reset parameters
+            _metric.set_parameters(mp)
+
+            # Deduce an isotropic metric from the error indicator field
+            _metric.compute_isotropic_metric(error_indicator=indi, interpolant="L2")
+            _metric.normalise()
+
+            # append the metric for the step in the time partition
+            _metrics.append(_metric)
+
+        # set the first metric to the base
+        metric = _metrics[0]
+        # all the other metrics
+        metric.average(*_metrics[1:], weights=[dt] * len(_metrics))
+
+        metrics.append(metric)
+
+    # Apply space time normalisation
+    space_time_normalise(metrics, mesh_seq.time_partition, mp)
+
+    # Adapt each mesh w.r.t. the corresponding metric, provided it hasn't converged
+    for i, metric in enumerate(metrics):
+        if not mesh_seq.converged[i]:
+            mesh_seq[i] = adapt(mesh_seq[i], metric)
+        complexities.append(metric.complexity())
+    num_dofs = mesh_seq.count_vertices()
+    num_elem = mesh_seq.count_elements()
+    pyrint(f"fixed point iteration {iteration + 1}:")
+    for i, (complexity, ndofs, nelem) in enumerate(
+        zip(complexities, num_dofs, num_elem)
+    ):
+        pyrint(
+            f"  subinterval {i}, complexity: {complexity:4.0f}"
+            f", dofs: {ndofs:4d}, elements: {nelem:4d}"
+        )
+
+    # Plot each intermediate adapted mesh
+    fig, axes = mesh_seq.plot()
+    for i, ax in enumerate(axes):
+        ax.set_title(f"Subinterval {i + 1}")
+    fig.savefig(f"burgers-hessian_mesh{iteration + 1}.jpg")
+    plt.close()
+
+    # Since we have two subintervals, we should check if the target complexity has been
```

Comment:
Maybe put 'multiple' rather than 'two', to keep things general.

Reply:
See issue: https://github.com/mesh-adaptation/animate/issues/84.